chia/timelord/timelord_api.py

```diff
@@ -99,13 +101,22 @@ async def new_peak_timelord(self, new_peak: timelord_protocol.NewPeakTimelord) -
             self.timelord.state_changed("skipping_peak", {"height": new_peak.reward_chain_block.height})
 
     def check_orphaned_unfinished_block(self, new_peak: NewPeakTimelord):
+        new_peak_unf_rh = new_peak.reward_chain_block.get_unfinished().get_hash()
         for unf_block in self.timelord.unfinished_blocks:
             if unf_block.reward_chain_block.total_iters <= new_peak.reward_chain_block.total_iters:
```

Comment:
it's not obvious to me why we should stick to our own unfinished block if the iteration count is equal.

Reply:
because someone is orphaning a block we already heard about that means it was propagated earlier, we shouldn't prefer the new one since we might be able to infuse the one we have and continue our fork, this was the initial motivation behind this change we introduced to avoid malicious orphaning